### PR TITLE
added register team and display participants, teams

### PR DESCRIPTION
**Pull request by arpit1997**
added requiremements.txt
added register team view
added team and participant display
added team profile vew
issue ref #6

### CODE REVIEW

**Comment by singhpratyush** [comment]
You were not supposed to work on team profile view, weren't you?

If you still wish to add the same, assign yourself #9 and work on team edit view too.

File: teambuilder/web/models.py [hunk 1]
```diff
@@ -9,6 +9,9 @@ class Participant(models.Model):
                                 on_delete=models.CASCADE)
     mobile = models.CharField(max_length=10, unique=True)
 
+    def get_full_name(self):
+        return self.user.first_name+self.user.last_name
```
Comment (singhpratyush) [inline teambuilder/web/models.py:13]: Shouldn't it be
```python
    return self.user.first_name + ' ' + self.user.last_name
```

File: teambuilder/web/templates/participant.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    {% if participants %}
+        <table>
+            {% for participant in participants %}
+                <tr>
+                    <td>
+                        {{ participant.get_full_name }}
+                    </td>
+                    <td>{{ participant.user.username }}</td>
+                    <td>{{ participant.mobile }}</td>
+                </tr>
```
Comment (singhpratyush) [inline teambuilder/web/templates/participant.html:15]: Add skills too.

File: teambuilder/web/templates/teams.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    {% if teams %}
+        <table>
+            {% for team in teams %}
+                <tr>
+                    <td>
+                        <a href="{% url 'team-profile' team.name %}">{{ team.name }}</a>
+                    </td>
+                    <td>{{ team.create_time }}</td>
+                    <td>{{ team.created_by.user.username }}</td>
+                </tr>
```
Comment (singhpratyush) [inline teambuilder/web/templates/teams.html:15]: Add overall skills and member count too.

File: teambuilder/web/urls.py [hunk 4]
```diff
@@ -3,7 +3,11 @@
...
-    url(r'^$', views.home),
-    url(r'^participant/register/?$', views.register),
-    url(r'^participant/login/?$', views.login),
+    url(r'^$', views.home, name='home'),
+    url(r'^participant/register/?$', views.register, name='register'),
+    url(r'^participant/login/?$', views.login, name='login'),
+    url(r'^team/new/?$', views.register_team, name='register-team'),
+    url(r'^team/?$', views.teams, name='view-teams'),
+    url(r'^participant/?$', views.participant, name='view-participants'),
+    url(r'^team/profile/(?P<team_name>[\S]+)/?$', views.team_profile, name='team-profile'),
```
Comment (singhpratyush) [inline teambuilder/web/urls.py:12]: Team names should be 4 to 64 chars long.
```python
    url(r'^team/profile/(?P<team_name>[a-zA-Z0-9\-]{4,64})/?$', views.team_profile, name='team-profile'),
```

File: teambuilder/web/views.py [hunk 5]
```diff
@@ -1,90 +1,155 @@
...
 
 def home(request):
-    user = utils.get_login_user(request.COOKIES)
-    if user is None:
-        return redirect('/participant/login')
-    return render(request, 'home.html', context={
-        'title': 'Welcome to Hackathon 2017',
-        'user': user,
-    })
+	user = utils.get_login_user(request.COOKIES)
```
Comment (singhpratyush) [inline teambuilder/web/views.py:14]: What is this diff for?

File: teambuilder/web/views.py [hunk 6]
```diff
@@ -1,90 +1,155 @@
...
+
+
+def register_team(request):
+	user = utils.get_login_user(request.COOKIES)
+	error = []
+	if user:
+		if request.POST.get('register', None):
+			team_name = request.POST.get('name')
+			# spaces are replaced by '-' due to security reasons
+			team_name = team_name.replace(' ', '-', 32)
```
Comment (singhpratyush) [inline teambuilder/web/views.py:104]: Not a good way. Just check team names for `[a-zA-Z0-9\-]{4,64}` and put up error if needed.

File: teambuilder/web/views.py [hunk 7]
```diff
@@ -1,90 +1,155 @@
...
+			description = request.POST.get('description')
+			created_by = user
+			created_time = timezone.now()
+			try:
+				with atomic():
+					team = models.Team.objects.create(name=team_name, created_by=created_by, create_time=created_time,
```
Comment (singhpratyush) [inline teambuilder/web/views.py:110]: `create_time` is managed internally by Django.

File: teambuilder/web/views.py [hunk 8]
```diff
@@ -1,90 +1,155 @@
...
+													  description=description)
+					team.save()
+					return HttpResponseRedirect('/')
```
Comment (singhpratyush) [inline teambuilder/web/views.py:113]: `django.shortcuts.redirect` is already imported.

File: teambuilder/web/views.py [hunk 9]
```diff
@@ -1,90 +1,155 @@
...
+			except Exception as e:
+				print(e)
+				error.append("failed to create team - {0}".format(e))
+	else:
+		error.append("Please login to create team")
+	return render(
+		request, 'register_team.html', context={
+			'title': 'register your team',
+			'errors': error,
+			'user': None,
```
Comment (singhpratyush) [inline teambuilder/web/views.py:123]: User will be required for headers here.

File: teambuilder/web/views.py [hunk 10]
```diff
@@ -1,90 +1,155 @@
...
+			try:
+				with atomic():
+					team = models.Team.objects.create(name=team_name, created_by=created_by, create_time=created_time,
+													  description=description)
+					team.save()
+					return HttpResponseRedirect('/')
+			except Exception as e:
+				print(e)
+				error.append("failed to create team - {0}".format(e))
+	else:
```
Comment (singhpratyush) [inline teambuilder/web/views.py:117]: Why not simply redirecting to `/team` in case he hasn't logged in.

File: teambuilder/web/views.py [hunk 11]
```diff
@@ -1,90 +1,155 @@
...
+			'teams': team,
+		})
+
+
+def team_profile(request, team_name):
+	try:
+		team = models.Team.objects.get(name=team_name)
+	except ObjectDoesNotExist:
+		print("No team found with this name")
+		return HttpResponse('404 not found')
```
Comment (singhpratyush) [inline teambuilder/web/views.py:141]: Use `404.html`.

**Review by singhpratyush — CHANGES_REQUESTED** [review]
Please consider the mentioned changes.

File: teambuilder/web/views.py (hunk 5 above)
Comment (arpit1997) [reply teambuilder/web/views.py:14]: This might be a mistake. I will correct it.

File: teambuilder/web/views.py (hunk 6 above)
Comment (arpit1997) [reply teambuilder/web/views.py:104]: What if a team name has spaces in it?
consider [this](http://stackoverflow.com/questions/497908/is-a-url-allowed-to-contain-a-space)

Comment (singhpratyush) [reply teambuilder/web/views.py:104]: If you look at the regex that I've written, it doesn't allows spaces or any character that may be escaped as URL.

Comment (arpit1997) [reply teambuilder/web/views.py:104]: When team name is for example `fire n ice` then what will be the team profile url ?
Aren't we allowing spaces in the team names?

Comment (singhpratyush) [reply teambuilder/web/views.py:104]: > Just check team names for `[a-zA-Z0-9\-]{4,64}`

If you look at the regex I'm talking about, we don't allow spaces.

**Pull request closed without merging**